pyiceberg/table/update/__init__.py

```diff
@@ -177,8 +177,20 @@ class RemovePropertiesUpdate(IcebergBaseModel):
 
 class SetStatisticsUpdate(IcebergBaseModel):
     action: Literal["set-statistics"] = Field(default="set-statistics")
-    snapshot_id: int = Field(alias="snapshot-id")
     statistics: StatisticsFile
+    snapshot_id: Optional[int] = Field(
+        None,
+        alias="snapshot-id",
+        description="This optional field is **DEPRECATED for REMOVAL** since it contains redundant information. Clients should use the `statistics.snapshot-id` field instead.",
```

Comment:
Let's make it more explicit!
```suggestion
        description="snapshot-id is **DEPRECATED for REMOVAL** since it contains redundant information. Use `statistics.snapshot-id` field instead.",
```